### PR TITLE
docs: add sphinx documentation

### PR DESCRIPTION
**Pull request by jia200x**
This PR adds initial support for Sphinx documentation.

I suggest to read [this](https://brendanhasz.github.io/2019/01/05/sphinx.html) to have a better understanding of the usage The root file Sphinx file is the `index.rst`, from there we can include other files and module documentation.

To build the documentation:
- `pipenv run doc`

This command will call Sphinx's `make html` command to build HTML documentation. Other documentation formats are also available (man pages, latex, JSON, XML, etc).

The documentation is placed under `_build/html/index.html`

This PR also adds CircleCI build tests for the documentation, so we ensure the documentation is always consistent.

We need to slowly migrate all functions to their Sphinx format.

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
> We need to slowly migrate all functions to their Sphinx format.

I suggest we make either a tracking issue or a project for this migration.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (sasalatart) [inline Makefile:1]: I am not very familiar with `Makefile`s. Is there any possibility of placing doc-related things inside a `docs` dir or something like that?

Having a `Makefile` in the root dir of the repo is a little bit strange. Someone from the outside seeing the Makefile in the root dir might think that the whole app is built using Make. If the answer is no, then it's no problem.

Comment (sasalatart) [reply Makefile:1]: And then somehow calling `make` from the `Pipfile`, but pointing to `docs/Makefile`

Comment (sasalatart) [reply Makefile:1]: (This is not a necessary change though, just a question)

Comment (leandrolanzieri) [reply Makefile:1]: I'm thinking if having this Makefile is actually really necessary. Perhaps we can just keep the commands on pipenv (as far as we can).

Comment (sasalatart) [reply Makefile:1]: Yeah, if that approach is possible I would vote for that!

Comment (jia200x) [reply Makefile:1]: We can get rid of the Makefile. I can do it :)

**Comment by felipehuerta17** [comment]
Hi @jia200x , I pulled your branch and tried

`pipenv run doc`

But I got the following error message:

image

Do you know what may be the reason? Should everyone receive that Theme error?

**Comment by felipehuerta17** [comment]
@sasalatart likes Sphinx because [this extension](https://brendanhasz.github.io/2019/01/05/sphinx.html#napoleon-extension)

**Comment by jia200x** [comment]
> @sasalatart likes Sphinx because this extension

Actually we should indeed use that one!

**Comment by jia200x** [comment]
@sasalatart it seems the CircleCI script is failing. Have you had an issue like that before?

**Comment by jia200x** [comment]
Just to save some time, I simply migrated the whole API to Sphinx.

The format of the "index.rst" is not the best, but it's a base for follow up work.
I also addressed the `pipenv run doc` request so now pipenv directly invokes the `sphinx-build` command. I removed the Makefile and make.bat files.

**Comment by jia200x** [comment]
let's wait for #26

**Comment by felipehuerta17** [comment]
@jia200x will rebasing this to master solve the ci/circleci complain?

**Pull request merged**